packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -2035,13 +2035,87 @@ void main() {
       await gesture.moveTo(textOffsetToPosition(paragraph3, 6));
       await gesture.up();
 
-      // keyboard copy.
+      // Send a keyboard copy..
+      await sendKeyCombination(tester, const SingleActivator(LogicalKeyboardKey.keyC, control: true));
+
+      final Map<String, dynamic> clipboardData = mockClipboard.clipboardData as Map<String, dynamic>;
+      expect(clipboardData['text'], 'w are you?Good, and you?Fine, ');
+    }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.android, TargetPlatform.windows, TargetPlatform.linux, TargetPlatform.fuchsia }));
+
+    testWidgets('can copy a selection and insert separator', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: SelectableRegion(
+            focusNode: FocusNode(),
+            selectionControls: materialTextSelectionControls,
+            child: const Column(
+              children: <Widget>[
+                Text('How are you?'),
+                Text('Good, and you?'),
+                Text('Fine, thank you.'),
+              ],
+            ),
+          ),
+        ),
+      );
+      // Select from offset 2 of paragraph 1 to offset 6 of paragraph3.
+      final RenderParagraph paragraph1 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('How are you?'), matching: find.byType(RichText)));
+      final TestGesture gesture = await tester.startGesture(textOffsetToPosition(paragraph1, 2), kind: PointerDeviceKind.mouse);
+      addTearDown(gesture.removePointer);
+      await tester.pump();
+
+      final RenderParagraph paragraph3 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Fine, thank you.'), matching: find.byType(RichText)));
+      await gesture.moveTo(textOffsetToPosition(paragraph3, 6));
+      await gesture.up();
+
+      // Send a keyboard copy..
       await sendKeyCombination(tester, const SingleActivator(LogicalKeyboardKey.keyC, control: true));
 
       final Map<String, dynamic> clipboardData = mockClipboard.clipboardData as Map<String, dynamic>;
       expect(clipboardData['text'], 'w are you?Good, and you?Fine, ');
     }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.android, TargetPlatform.windows, TargetPlatform.linux, TargetPlatform.fuchsia }));
 
+    testWidgets('can change copyIntercept with CopyInterceptor', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: SelectableRegion(
+            focusNode: FocusNode(),
+            selectionControls: materialTextSelectionControls,
+            copyIntercept: CopyIntercept.newline,
+            child: const Column(
+              children: <Widget>[
+                Text('How are you?'),
+                CopyInterceptor(
```

Comment:
I wonder if the `copyInterceptor` functionality should go under `DefaultSelectionStyle`. Then the user can wrap their subtree with `DefaultSelectionStyle.merge(copyInterceptor: CopyIntercept.space)` to override only the `copyInterceptor`.

Overall, I like this approach since it is able to propagate the `copyInterceptor` down a subtree. I'm just weary about adding a new widget to support this kind of functionality since it feels like this should be part of some overall selection configuration.

Reply:
Hi @Rexios80, wanted to follow up on this comment. Do you have any thoughts on this approach? I think the `SelectionContainer` would be the one inheriting the copyInterceptor from its nearest `DefaultSelectionStyle`.

Reply:
How will that work for defaults on a per-widget basis? Would they usa a `DefaultSelectionStyle` in their `build` method unless there is a parent `DefaultSelectionStyle`?

Reply:
If I understand correctly, a descendant widget of a `DefaultSelectionStyle` would use `DefaultSelectionStyle.of(context)` to inherit any ancestor configuration, or use a local `DefaultSelectionStyle` to override any ancestor configuration.

Given `SelectableRegion` it should have a `DefaultSelectionStyle.merge` in its build method defining a default (probably newline). And then descendant widgets can override this. Let me know if this is what you were asking.